src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1090,6 +1089,22 @@ lemma map_is_path_iff_of_injective (hinj : function.injective f) :
   (p.map f).is_path ↔ p.is_path :=
 ⟨is_path.of_map, map_is_path_of_injective hinj⟩
 
+lemma map_is_cycle_of_injective {u : V} {p : G.walk u u}
+  (hinj : function.injective f) (hp : p.is_cycle) : (p.map f).is_cycle :=
+begin
+  cases p,
+  { simp only [is_cycle.not_of_nil] at hp,
+    exact hp.elim, },
+  { simp only [cons_is_cycle_iff, map_cons, edges_map, list.mem_map, not_exists, not_and] at hp ⊢,
+    refine ⟨map_is_path_of_injective hinj hp.left, _⟩,
+    rintro ⟨a, b⟩ mem edge_eq,
+    change sym2.map f.to_fun ⟦⟨a, b⟩⟧ = ⟦(f.to_fun u, f.to_fun p_v)⟧ at edge_eq,
+    simp only [sym2.map_pair_eq, rel_hom.coe_fn_to_fun, quotient.eq, sym2.rel_iff] at edge_eq,
+    rcases edge_eq with (⟨au, bv⟩|⟨av, bu⟩),
+    { cases hinj au, cases hinj bv, exact hp.right mem, },
+    { cases hinj av, cases hinj bu, rw sym2.eq_swap at hp, apply hp.right mem, }, },
+end
```

Comment:
Please replace this with

```lean
lemma map_is_trail_iff_of_injective (hinj : function.injective f) :
  (p.map f).is_trail ↔ p.is_trail :=
begin
  induction p with w u v w huv hvw ih,
  { simp },
  { rw [map_cons, cons_is_trail_iff, cons_is_trail_iff, edges_map],
    change _ ∧ sym2.map f ⟦(u, v)⟧ ∉ _ ↔ _,
    rw list.mem_map_of_injective (sym2.map.injective hinj),
    exact and_congr_left' ih, },
end

alias map_is_trail_iff_of_injective ↔ _ map_is_trail_of_injective

lemma map_is_cycle_iff_of_injective {p : G.walk u u} (hinj : function.injective f) :
  (p.map f).is_cycle ↔ p.is_cycle :=
by rw [is_cycle_def, is_cycle_def, map_is_trail_iff_of_injective hinj, ne.def, map_eq_nil_iff,
       support_map, ← list.map_tail, list.nodup_map_iff hinj]

alias map_is_cycle_iff_of_injective ↔ _ map_is_cycle_of_injective
```

Reply:
oh, that's efficient!

Reply:
I forgot to give you this too:
```lean
@[simp] lemma map_eq_nil_iff {p : G.walk u u} : p.map f = nil ↔ p = nil :=
by cases p; simp
```
You could put it right before or after `map_eq_of_eq`